doc/whats_new/v0.20.rst

```diff
@@ -504,6 +504,10 @@ Preprocessing
   when returning a sparse matrix output. :issue:`11042` by :user:`Daniel
   Morales <DanielMorales9>`.
 
+- Fix inconsistency between sparse and dense case in
```

Comment:
Is this merely an API inconsistency, or will it affect users?

Reply:
When running twice `fit` with sparse matrix and `with_mean=False` and `with_std=False`, it was crashing. That's why I thought it would be a bug fix.

Do you prefer to move it to API change summary (or not document it in what's new?)

Reply:
and in the dense case we change `mean_` from returning an array to returning `None` which seems more logic and what is already happening for `var_`.

Reply:
I'm just trying to work out how to make clear to users how this change will affect them. I don't think your message makes that clear at all.

Reply:
Oh ok, is this formulation better

```
Fix inconsistencies in :class:`preprocessing.StandardScaler` with `with_mean=False`
and `with_std=False`. ``mean_`` will be set to ``None`` with both sparse and dense
inputs. ``n_samples_seen_`` will be also reported for both input types.
```